pkg/probe/http/http.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/probe"
+
+	"github.com/golang/glog"
+)
+
+var client = &http.Client{}
```

Comment:
Can we not do globals?  What if each of these probe sub-pkgs had a New() method that returned a pkg-specific type which implemented their particular Probe() method.  httpprobe.T.New() would take a *http.Client, and use that when Probe() was called.

kubelet would have to call this New and save the results somewhere.

It's not a HUGE deal, but we've been bitten by globals a few times now, in things like monolith binaries and tests.